src/opencensus.erl

```diff
@@ -103,23 +104,41 @@ start_span(_Name, undefined) ->
     undefined;
 start_span(Name, #trace_context{trace_id=TraceId,
                                 span_id=ParentId}) ->
-    start_span(Name, TraceId, ParentId);
+    start_span(Name, TraceId, ParentId, maps:new());
 start_span(Name, #span{trace_id=TraceId,
                        span_id=ParentId}) ->
-    start_span(Name, TraceId, ParentId).
+    start_span(Name, TraceId, ParentId, maps:new()).
 
--spec start_span(unicode:unicode_binary(), maybe(integer()), maybe(integer())) -> maybe(span()).
+-spec start_span(unicode:unicode_binary(), maybe(trace_context() | span() | integer()), maybe(integer()) | map()) -> maybe(span()).
```

Comment:
I'm not a big fan of overloading method arguments. It makes code harder to read and more prone to mistakes. I'd prefer only allowing maps in the 4 argument version.

Reply:
Hm, what about just removing the `start_span(Name, TraceId, ParentId)` version? Require that to alawys be `start_span(Name, TraceId, ParentId, Attributes)`? So its:

```
-spec start_span(unicode:unicode_binary(), maybe(trace_context() | span())) -> maybe(span()). 

-spec start_span(unicode:unicode_binary(), maybe(trace_context() | span()), map()) -> maybe(span()). 

-spec start_span(unicode:unicode_binary(), trace_id(), span(), map()) -> maybe(span()). 
```

Reply:
I'd not have a problem with that. My issue was that the same rarity of a call could mean two very different things based on the 'type' of the arguments.

Reply:
@deadtrickster thoughts?